apps/dotcom/sync-worker/src/worker.ts

```diff
@@ -42,15 +43,18 @@ const router = createRouter<Environment>()
 		joinExistingRoom(req, env, ROOM_OPEN_MODE.READ_ONLY)
 	)
 	.get('/app/file/:roomId', forwardRoomRequest)
-	.get('/app/:userName', async (req, env) => {
-		const userName = req.params.userName
-		if (!userName) return notFound()
+	.get('/app', async (req, env) => {
+		const auth = await getAuth(req, env)
+		if (!auth?.userId) return notFound()
 
 		// This needs to be a websocket request!
 		if (req.headers.get('upgrade')?.toLowerCase() === 'websocket') {
 			// Set up the durable object for this room
-			const id = env.TLAPP_DO.idFromName(userName)
-			return env.TLAPP_DO.get(id).fetch(req)
+			const id = env.TLAPP_DO.idFromName(auth.userId)
+			const url = new URL(req.url)
+			url.pathname = `/app/${auth.userId}`
```

Comment:
it'd be great (in a followup is fine) to have a `urlFactory` place where the `/app` and related paths are in one place, we'll start to lose track.

`dotcom-shared/src/routes.ts` would make sense for this.